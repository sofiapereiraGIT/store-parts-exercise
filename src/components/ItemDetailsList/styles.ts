import styled from "styled-components";

export const ItemDetailsList = {
  DetailsList: styled.div`
    margin-top: 2.5em;
  `,
  DetailsRow: styled.div`
    margin-top: 0.625em;
  `,
  DetailsField: styled.span`
    font-weight: bold;
    font-size: 1.17em;
    text-transform: capitalize;
  `,
};
