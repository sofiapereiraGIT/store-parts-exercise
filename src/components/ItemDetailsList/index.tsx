import { useHistory } from "react-router-dom";
import { Item } from "../../models/item";
import { ItemDetailsList as S } from "./styles";

export default function ItemDetailsList() {
  const history = useHistory();
  const state: Item | undefined = history.location?.state as Item;
  const item: Item | undefined = state && {
    name: state.name,
    type: state.type,
    price: state.price,
  };

  return (
    <S.DetailsList>
      {item ? (
        Object.entries(item).map(([key, value]) => (
          <S.DetailsRow key={key}>
            <S.DetailsField>{key}:</S.DetailsField> <span>{value}</span>
          </S.DetailsRow>
        ))
      ) : (
        <S.DetailsRow>No item loaded</S.DetailsRow>
      )}
    </S.DetailsList>
  );
}
