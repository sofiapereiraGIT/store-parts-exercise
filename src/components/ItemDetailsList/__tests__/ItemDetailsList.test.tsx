import React from "react";
import renderer from "react-test-renderer";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";
import ItemDetailsList from "../index";

describe("ItemDetailsList Tests", () => {
  it("Renders properly when an item is provided", () => {
    const history = createMemoryHistory({
      initialEntries: [
        {
          state: {
            name: "itemName",
            type: "itemType",
            price: "itemPrice",
          },
        },
      ],
    });
    const renderedComponent = renderer.create(
      <Router history={history}>
        <ItemDetailsList />
      </Router>
    );
    expect(renderedComponent.toJSON()).toMatchSnapshot();
  });
});
