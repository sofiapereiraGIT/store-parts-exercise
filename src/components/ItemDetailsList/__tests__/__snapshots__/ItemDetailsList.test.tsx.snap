// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetailsList Tests Renders properly when an item is provided 1`] = `
<div
  className="sc-bczRLJ gtXPJG"
>
  <div
    className="sc-gsnTZi eDBMpS"
  >
    <span
      className="sc-dkzDqf iBjklj"
    >
      name
      :
    </span>
     
    <span>
      itemName
    </span>
  </div>
  <div
    className="sc-gsnTZi eDBMpS"
  >
    <span
      className="sc-dkzDqf iBjklj"
    >
      type
      :
    </span>
     
    <span>
      itemType
    </span>
  </div>
  <div
    className="sc-gsnTZi eDBMpS"
  >
    <span
      className="sc-dkzDqf iBjklj"
    >
      price
      :
    </span>
     
    <span>
      itemPrice
    </span>
  </div>
</div>
`;
